src/process_utility.c

```diff
@@ -2494,6 +2495,14 @@ process_index_chunk_multitransaction(int32 hypertable_id, Oid chunk_relid, void
 	}
 #endif
 
+	chunk = ts_chunk_get_by_relid(chunk_relid, true);
+	if (IS_OSM_CHUNK(chunk)) /*cannot create index on foreign OSM chunk */
```

Comment:
What about add a NOTICE message where about skipping?

Reply:
By moving `chunk = ts_chunk_get_by_relid(chunk_relid, true);` above you're breaking the lock semantics described below:
```
...
	 * Hold a lock on the hypertable index, and the chunk to prevent
	 * from being altered. Since we use the same relids across transactions,
...
```

Reply:
I don't really know.

Reply:
Seems this is to prevent a concurrent session alter the same object (alter/drop columns, add/drop indexes, etc). Will be better @gayyappan to move your check and skip to after `table_open` and `index_open`

Reply:
@gayyappan is there any problems with moving this block back into the lock-protected section?

Patch: https://gist.github.com/zilder/a9b2b61af07361ae6a938b75ae516fb2

Reply:
yes, it  is ok to move the check after the table_open. This code block will not work as-is. Note that we should clear state correctly before exiting from the function. Probably easiest with an if-else stmt.

Reply:
moved the code around to address the problem described by mfundul.